packages/connect/src/protocol/async-iterable.ts

```diff
@@ -1148,6 +1149,42 @@ function parseLengthHint(
   return [true, n];
 }
 
+/**
+ * Wait for the first element of an iterable without modifying the iterable.
+ * This consumes the first element, but pushes it back on the stack.
+ *
+ * @private Internal code, does not follow semantic versioning.
+ */
+export async function untilFirst<T>(
```

Comment:
suggestion: of course, we're exercising this indirectly by the means of the other tests in this PR, but it'd be super great to have some coverage for this function because it's a bit esoteric what it actually does and if we ever have to come back here and adjust it it'd be pretty hard.

If you're interested: I'd love to help with writing that test if you're interested in pairing on it!

Reply:
Won't say no to a couple of concise tests for this, but we'll get better mileage from improving the crosstests. Not saying that they are a replacement unit tests for this function, but that there is more pressing need.